regression/test.pl

```diff
@@ -64,9 +64,18 @@ ($)
   return @data;
 }
 
-sub test($$$$$$) {
-  my ($name, $test, $t_level, $cmd, $ign, $dry_run) = @_;
+sub test($$$$$$$) {
+  my ($name, $test, $t_level, $cmd, $ign, $dry_run, $definition) = @_;
   my ($level, $input, $options, $grep_options, @results) = load("$test");
+  my $equal_sign = index($definition, '=');
+  if ($equal_sign != -1) {
+    my $key = substr($definition, 0, $equal_sign);
+    my $value = substr($definition, $equal_sign + 1);
+    $options =~ s/\$$key/$value/g;
+    if (index($options, " \$") != -1 || (substr($options, 0, 1) cmp '$') == 0) {
+      print "\n  Warning: Key \"$key\" not found\n";
```

Comment:
Could the message rather be:
`Warning: a '$' symbol has been found in the arguments but no corresponding '-D' option exists`?